lib/utils/validateTypes.mjs

```diff
@@ -118,3 +120,45 @@ export function assertString(value) {
 	// eslint-disable-next-line no-console
 	console.assert(isString(value), `"${value}" must be a string`);
 }
+
+/**
+ * @param {unknown} value
+ * @returns {value is (Range['start'] | Range['end'])}
+ */
+export function isPosition(value) {
+	if (!isPlainObject(value)) return false;
+
+	if (!isNumber(value.line)) return false;
+
+	if (!isNumber(value.column)) return false;
+
+	return true;
+}
+
+/**
+ * @param {unknown} value
+ * @returns {value is Range}
+ */
+export function isRange(value) {
+	if (!isPlainObject(value)) return false;
+
+	if (!isPosition(value.start)) return false;
+
+	if (!isPosition(value.end)) return false;
+
+	return true;
+}
+
+/**
+ * @param {unknown} value
+ * @returns {value is { index: number, endIndex: number }}
+ */
+export function hasIndices(value) {
```

Comment:
[suggestion] `hasIndices` seems too specific to be added to this `validateTypes` since the name starts with `has`, not `is`.

Instead, how about moving `hasIndices` to `report.mjs` as a private function?